lib/bang.coffee

```diff
@@ -129,6 +133,7 @@ module.exports = Bang =
         atom.notifications.addSuccess(dirMessage, {detail: output, dismissable: missNote})
       else
         range = editor.getSelectedBufferRange()
+        if range.isEmpty() then range = editor.getBuffer().getRange()
```

Comment:
This line will replace the whole buffer if no text is selected even if you want to add the output of the command to your buffer.